onetl/connection/file_connection/samba.py

```diff
@@ -0,0 +1,306 @@
+#  Copyright 2023 MTS (Mobile Telesystems)
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+from __future__ import annotations
+
+import os
+import stat
+import textwrap
+from io import BytesIO
+from logging import getLogger
+from typing import Optional
+
+from etl_entities.instance import Host
+from pydantic import SecretStr, validator
+
+from onetl.connection.file_connection.file_connection import FileConnection
+from onetl.hooks import slot, support_hooks
+from onetl.impl import LocalPath, RemotePath, RemotePathStat
+
+try:
+    from smb.base import NotConnectedError
+    from smb.smb_structs import OperationFailure
+    from smb.SMBConnection import SMBConnection
+except (ImportError, NameError) as e:
+    raise ImportError(
+        textwrap.dedent(
+            """
+            Cannot import module "pysmb".
+
+            Since onETL v0.7.0 you should install package as follows:
+                pip install onetl[samba]
+
+            or
+                pip install onetl[files]
+            """,
+        ).strip(),
+    ) from e
+
+
+log = getLogger(__name__)
+
+
+@support_hooks
+class Samba(FileConnection):
+    """Samba file connection.
+
+    Based on `pysmb library <https://pypi.org/project/pysmb/>`_.
+
+    Parameters
+    ----------
+    host : str
+        Host of Samba source. For example: ``msk.mts.ru``.
```

Comment:
```suggestion
        Host of Samba source. For example: ``mydomain.com``.
```
Внутренний адрес, не стоит такое делать публичным